msticpy/context/azure/sentinel_core.py

```diff
@@ -35,6 +39,59 @@
 
 logger = logging.getLogger(__name__)
 
+_SUB_ID = "subscription_id"
+_RES_GRP = "resource_group"
+_WS_NAME = "workspace_name"
+_RES_ID = "resource_id"
+
+
+@dataclass
+class SentinelInstanceDetails:
+    """Dataclass for Sentinel workspace details."""
+
+    subscription_id: Optional[str]
+    resource_group: Optional[str]
+    workspace_name: Optional[str]
+
+    @property
+    def resource_id(self) -> Optional[str]:
+        """Return the resource ID for the workspace."""
+        if not all([self.subscription_id, self.resource_group, self.workspace_name]):
+            return None
+        return build_sentinel_resource_id(
+            self.subscription_id, self.resource_group, self.workspace_name  # type: ignore
+        )
+
+
+def _get_ws_defaults(
+    *,
+    subscription_id: Optional[str] = None,
+    resource_group: Optional[str] = None,
+    workspace_name: Optional[str] = None,
+) -> SentinelInstanceDetails:
+    """Return default values for Sentinel workspace settings."""
+    return SentinelInstanceDetails(
+        subscription_id=subscription_id,
+        resource_group=resource_group,
+        workspace_name=workspace_name,
+    )
+
+
+_LEGACY_PARAM_NAMES = {
+    "sub_id": _SUB_ID,
+    "res_grp": _RES_GRP,
+    "ws_name": _WS_NAME,
+    "workspace": _WS_NAME,
+    "res_id": _RES_ID,
+}
+_CORE_WS_PARAMETERS = [_SUB_ID, _RES_GRP, _WS_NAME]
+_WS_PARAMETERS = _CORE_WS_PARAMETERS + [_RES_ID]
+
+
+def _map_legacy_param_names(**kwargs) -> Dict[str, Any]:
+    """Map legacy parameter names to current names."""
```

Comment:
It might be worth expanding on this docstring because of the heavy lifting that this function is going:

- Historically, msticpy used abbreviated versions of the kwarg parameters needed to connect to a Sentinel instance
- These names have since changed, but we want to accept the old kwargs and map them behind the scenes to their new names
- This function will search through the passed-in kwargs to find these outdated kwarg names and replace them with their modern equivalents. If not found, then it simply passes the kwarg along as-is.